sideloader/tests/test_sideloader.py

```diff
@@ -0,0 +1,521 @@
+import os
+
+import pytest
+
+from sideloader import Build, Deploy, GitRepo, Package, Workspace
+from sideloader.config_files import ConfigFiles
+from sideloader.deploy_types import DeployType
+
+
+class TestGitRepo(object):
+    def test_from_github_url(self):
+        repo = GitRepo.from_github_url(
+            'https://github.com/praekelt/sideloader2.git', 'develop')
+
+        assert repo.url == 'https://github.com/praekelt/sideloader2.git'
+        assert repo.name == 'sideloader2'
+        assert repo.branch == 'develop'
+
+
+class CommandLineTest(object):
+    def setup_method(self, test_method):
+        self.cmds = []
+
+    def cmd(self, args, *_args, **kwargs):
+        self.cmds.append(args)
+
+
+class TestWorkspace(CommandLineTest):
+
+    def _create_workspace(self, tmpdir):
+        repo = GitRepo('https://github.com/praekelt/sideloader2.git',
+                       'develop', 'sideloader2')
+
+        workspace = Workspace('test_id', str(tmpdir), '/var/praekelt', repo)
+        workspace._cmd = self.cmd
+        return workspace
```

Comment:
You can probably wrap this in a pytest fixture to avoid calling it in every test: http://pytest.org/latest/fixture.html